### PR TITLE
feat: [PL-23816]: Redirect to login when ip not whitelisted

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
##### Screenshots:
Screenshot 2022-05-18 at 7 56 09 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/framework/app/App.tsx [hunk 1]
```diff
@@ -143,6 +144,15 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
       }
       global401HandlerUtils(history)
       return
+    } else if (response.status === 400) {
+      response.json().then(res => {
+        const notWhiteListedMessage =
+          res?.responseMessages?.find((message: any) => message?.code === 'NOT_WHITELISTED_IP') || undefined
+        if (notWhiteListedMessage) {
+          showError('Current IP Address is not whitelisted.')
```
Comment (abhinavrastogi-harness) [inline src/framework/app/App.tsx:152]: does the response contain the ip address? we should add it to the message.

Comment (abhinavrastogi-harness) [reply src/framework/app/App.tsx:152]: also, can we use strings framework here?

File: src/framework/app/App.tsx [hunk 2]
```diff
@@ -143,6 +144,15 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
       }
       global401HandlerUtils(history)
       return
+    } else if (response.status === 400) {
+      response.json().then(res => {
```
Comment (abhinavrastogi-harness) [inline src/framework/app/App.tsx:148]: we should add a rejection handled incase the json parsing fails.. it'll go to the second callback param of .then()

File: src/framework/app/App.tsx [hunk 3]
```diff
@@ -143,6 +144,15 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
...
+        const notWhiteListedMessage =
+          res?.responseMessages?.find((message: any) => message?.code === 'NOT_WHITELISTED_IP') || undefined
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply src/framework/app/App.tsx:150]: removed

File: src/framework/app/App.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/framework/app/App.tsx:152]: Message was coming in response, didn't realise it.
changed it now

Comment (avinashmadhwani02) [reply src/framework/app/App.tsx:152]: we can't use string framework inside App.tsx. other messages are also hard coded here

File: src/framework/app/App.tsx [hunk 4]
```diff
@@ -143,6 +144,15 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
       }
       global401HandlerUtils(history)
       return
+    } else if (response.status === 400) {
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply src/framework/app/App.tsx:147]: done

File: src/framework/app/App.tsx [hunk 5]
```diff
@@ -123,28 +124,59 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
...
+            })
+            .catch(() => {
+              window.bugsnagClient?.notify?.(
+                new Error('Error handling 400 status code'),
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                function (event: any) {
+                  event.severity = 'error'
+                  event.setUser(username)
+                  event.addMetadata('400 Details', {
+                    url: response.url,
```
Comment (abhinavrastogi-harness) [inline src/framework/app/App.tsx:170]: you can add the response string here to make the error more helpful

File: src/framework/app/App.tsx [hunk 6]
```diff
@@ -123,28 +124,59 @@ export function AppWithAuthentication(props: AppProps): React.ReactElement {
...
+            .then(res => {
+              const notWhiteListedMessage = res?.responseMessages?.find(
+                (message: any) => message?.code === 'NOT_WHITELISTED_IP'
+              )
+              if (notWhiteListedMessage) {
+                showError(notWhiteListedMessage.message)
+                global401HandlerUtils(history)
+              }
+            })
+            .catch(() => {
```
Comment (abhinavrastogi-harness) [inline src/framework/app/App.tsx:162]: shouldn't we use the second param of `.then()` instead of .catch?

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/framework/app/App.tsx (hunk 6 above)
Comment (avinashmadhwani02) [reply src/framework/app/App.tsx:162]: Will it catch json parsing error?

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**